base/broadcast.jl

```diff
@@ -313,7 +301,7 @@ ziptype{T}(::Type{T}, A) = typestuple(T, A)
 ziptype{T}(::Type{T}, A, B) = (Base.@_pure_meta; Iterators.Zip2{typestuple(T, A), typestuple(T, B)})
 @inline ziptype{T}(::Type{T}, A, B, C, D...) = Iterators.Zip{typestuple(T, A), ziptype(T, B, C, D...)}
 
-_broadcast_type{S}(::Type{S}, f, T::Type, As...) = Base._return_type(S, typestuple(S, T, As...))
+_broadcast_type{S}(::Type{S}, f, T::Type, As...) = Base._return_type(f, typestuple(S, T, As...))
```

Comment:
@TotalVerb I'm not sure what the purpose of the first type argument is but it seemed wrong to pass it as the first argument to `_return_type` and it also caused a test failure in the test at line 366 below. Does this change seem right to you?